src/App.js

```diff
@@ -18,28 +22,115 @@ console.log(initialStoreItems)
 
 export default function App() {
   // Setup state here...
+const [storeItem, setStoreItem] = useState(initialStoreItems)
+const [cartItems, setCartItems] = useState ([])
+
+
+// by using the button "add to cart"
+// find the item from the storeitems
+// and than add the item to the cart
+// by clicking button everytime it has to add new item to the cart.
+const addItemCart = item => {
+const itemsInStore = storeItem.find(items =>
+  {
+    return items.id === item.target.id
+  }) 
+  const itemInCart = cartItems.find(itemInCart => {
+      return itemInCart.id === item.target.id
+      })
+      if (itemInCart) {
+        itemInCart.quantity++
+            setCartItems([...cartItems])
+          } else {
+            let newCartItem = { ...itemsInStore, quantity: 1 }
+            setCartItems([...cartItems, newCartItem])
+          }
+}
+
+const CalculateCartTotal = () => {
+  let currentTotal = 0
+  cartItems.forEach(itemInCart => {
+    currentTotal += itemInCart.quantity * itemInCart.price
+  })
+console.log(currentTotal)
+  return currentTotal.toFixed(2)
+}
+  //  increase the quantity of items
+const handleIncrement = cartItem => {
+
+  const newCartItems = cartItems.map(targetCartItem => {
+    if (targetCartItem.name === cartItem.name) {
+    console.log("increase item", targetCartItem)
+      return {...targetCartItem, quantity: targetCartItem.quantity +1}
+    } else {
+      return targetCartItem
+    }
+  })
+ 
+  setCartItems(newCartItems)
+}
+// decrement button
+// check if the quqntity is more than 1 ,implement decrese 1
+// otherwise if it is equal to 1 remove from cart
+const handleDecrement =cartItem =>{
+  if(cartItem.quantity > 1) {
+    const newCartItems = cartItems.map(targetCartItem => {
+      if(targetCartItem.name === cartItem.name && targetCartItem.quantity) {
+        return {...targetCartItem, quantity: targetCartItem.quantity -1}
+      }
+      return targetCartItem
+    })
+    setCartItems(newCartItems)
+   
+  }else {
+    const removeItem = cartItems.filter(targetCartItem =>
+      targetCartItem.name !== cartItem.name)
+      setCartItems(removeItem)
+  
+  }
+}
 
   return (
     <>
       <header id="store">
         <h1>Greengrocers</h1>
         <ul className="item-list store--item-list">
           {/* Wrtite some code here... */}
+        {storeItem.map(items =>
+          (<StoreItmes
```

Comment:
You could pass the whole item rather than individual parts of it to make a bit tidier.